server/controllers/payroll/multiplePayroll/index.js

```diff
@@ -0,0 +1,1177 @@
+
+          // Annual cumulation of Base IPR
+          const annualCumulation = baseIpr * 12;
+
+          let ind = -1;
+          let iprValue = 0;
+          let scaleIndice;
+
+          if (iprScales.length) {
+            iprScales.forEach(scale => {
+              ind++;
+              if (annualCumulation > scale.tranche_annuelle_debut && annualCumulation <= scale.tranche_annuelle_fin) {
+                scaleIndice = ind;
+              }
+            });
+
+            const initial = iprScales[scaleIndice].tranche_annuelle_debut;
+            const rate = iprScales[scaleIndice].rate / 100;
+
+            const cumul = (iprScales[scaleIndice - 1]) ? iprScales[scaleIndice - 1].cumul_annuel : 0;
+            iprValue = (((annualCumulation - initial) * rate) + cumul) / 12;
+
+            if (nbChildren > 0) {
+              iprValue -= (iprValue * (nbChildren * 2)) / 100;
+            }
+
+            // Convert IPR value in selected Currency
+            iprValue = util.roundDecimal(iprValue * (enterpriseExchangeRate / iprExchangeRate), DECIMAL_PRECISION);
+
+            if (taxesContributions.length) {
+              taxesContributions.forEach(taxContribution => {
+              // Set the result of IPR calculation
+                if (taxContribution.is_ipr) {
+                  taxContribution.result = iprValue;
+                }
+
+                // Calculation of the sum of taxes and membership fee borne by the employee
+                if (taxContribution.is_employee) { sumTaxContributionEmp += taxContribution.result; }
+
+                allRubrics.push([uid, taxContribution.rubric_payroll_id, taxContribution.result]);
+              });
+            }
+          } else if (taxesContributions.length) {
+            taxesContributions.forEach(taxContribution => {
+              // Calculation of the sum of taxes and membership fee borne by the employee
+              if (taxContribution.is_employee) {
+                sumTaxContributionEmp += taxContribution.result;
+              }
+
+              allRubrics.push([uid, taxContribution.rubric_payroll_id, taxContribution.result]);
+            });
+
+          }
+
+          const netSalary = grossSalary - sumTaxContributionEmp;
+
+          const paiementData = {
+            uuid : uid,
+            employee_uuid : db.bid(employee.uuid),
+            payroll_configuration_id : payrollConfigurationId,
+            currency_id : data.currency_id,
+            basic_salary : basicSalary,
+            base_taxable : baseTaxable,
+            daily_salary : dailySalary,
+            total_day : totalDayPeriod,
+            working_day : data.working_day,
+            gross_salary : grossSalary,
+            net_salary : netSalary,
+            amount_paid : 0,
+            status_id : 2,
+          };
+
+          const deletePaiementData = 'DELETE FROM paiement WHERE employee_uuid = ? AND payroll_configuration_id = ?';
```

Comment:
Why are we deleting payment data ..?

Reply:
The payment configuration can be done several times and in each configuration it replaces the old Employee Payment and Payroll configuration data.